misc/python/materialize/scalability/operation.py

```diff
@@ -0,0 +1,30 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from typing import Optional
+
+from psycopg import Cursor, ProgrammingError
+
+
+class Operation:
+    def __init__(self) -> None:
+        self._query: Optional[str] = None
+
+    def execute(self, cursor: Cursor) -> None:
+        if self._query is None:
+            self._query = self.query()
+
+        try:
+            cursor.execute(self._query)
+            cursor.fetchall()
```

Comment:
We do not care about the result, that is why we discard it. At the same time, we need to run `fetchall()` in case any processing happens when fetching and not when executing -- depending on what technical direction the database takes, it may turn out that `execute()` is fast but the first time you fetch a row from your query a bunch of other stuff happens.